plaidcloud/utilities/frame_manager.py

```diff
@@ -2397,34 +2401,56 @@ def excel_to_csv_xlrd(excel_file_name, csv_file_name, sheet_name='sheet1', clean
             # The check on the first element of the tuple is to account for times.
             if skip_rows > 0 and skip_rows > rownum:
                 continue
-            if rownum == skip_rows and has_header:
+            if rownum == skip_rows:
                 # This is the header row. Force to clean header values
                 # Remove whitespace on either side
                 # Remove newlines
                 # Remove carriage returns
                 # Force to string
                 unique_column_names = set()
+                unique_source_names = set()
                 invalid_characters = set(',.()[];:*')
                 header_columns = []
 
-                for col in range(0, sh.ncols):
+                # logger.info(f'----- Detecting Column Headers on row: {rownum}')
+
+                for col in range(0, column_count):
                     # See if this column name is a date type
-                    column_name_dtype = dtype_from_excel(sh.cell(skip_rows, col).ctype)
-                    if column_name_dtype == 'timestamp':
-                        # Need to convert this from an Excel date stamp to human readable form
-                        column_date_name = xlrd.xldate_as_datetime(sh.cell(skip_rows, col).value, 0).date().isoformat()
-                        column_name = f'_{column_date_name}' # column names can't start with numbers so we prefix with _
-                    elif column_name_dtype == 'boolean':
-                        if sh.cell(skip_rows, col).value:
-                            column_name = 'TRUE'
+                    column_name_dtype = dtype_from_excel(sh.cell(rownum, col).ctype)
```

Comment:
Maybe grab the cell here, like below where it reads data `c = sh.cell(rownum, col_pos)`, then the code reads similar?`c.ctype, c.value`